### PR TITLE
Week 1 yuehan lee

### PR DESCRIPTION
**Pull request by lanyuke**
# Submitting Week 1 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅ ] Coding Challenges
- [✅ ] Coding Concepts
- [✅ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
The assessment is hard! But I like it!

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -9,61 +9,139 @@
...
+
+// Function name:boilingPointDeterminer
+const boilingPointDterminer = (num) => {
+if(num < 212){
+    return ` ${num} is below boiling point`
+} else if (num === 212){
+    return ` ${num} is at boiling point`
+} else {
+    return ` ${num} is above boiling point`
+}
```
Comment (chrisyhlee) [inline code-challenges.js:37]: Nice job! Make sure your indentation and spacing is uniform throughout.

File: code-challenges.js [hunk 2]
```diff
@@ -9,61 +9,139 @@
...
 // --------------------2) Create the code that will combine the two arrays and return the length using the test variables provided below.
 
 // Pseudo code:
+// input:arrays1, arrays2
+// output:return length value of combine arrays.
+// (possible solution: create a function to combine arrays1 and arrays2 and using .length method to get the length value)
+// 
+const lengthCombine =(arrays1, arrays2) =>{
+return arrays1.concat(arrays2).length
+}
```
Comment (chrisyhlee) [inline code-challenges.js:64]: Very nice! Same note as above, fix the indentation!

File: code-challenges.js [hunk 3]
```diff
@@ -9,61 +9,139 @@
...
 // Pseudo code:
+// Input: string1
+// Output: return reverse letters with string1.
+// (Possible solution is to create a function to use .split("") method to turn string1 into a array and use .reverse() method with to reverse letters then use .join("") method to turn it back to strings.)
+// function: reverseFunction
+
+
+const reverseFunction =(string1) => {
+    return (string1.split("").reverse().join(""))
+}
```
Comment (chrisyhlee) [inline code-challenges.js:86]: Woohoo! Indentation looks great, now just make sure you get your spacing down! Also, no need to wrap the whole return in parentheses! Well done!

File: code-challenges.js [hunk 4]
```diff
@@ -9,61 +9,139 @@
...
 // Pseudo code:
+// Input: array1, num1
+// Output: return last index of num1 in array1.
+// (possible solution is to create a function to use the .lastOfIndex method on array1)
+// Function: lastIndexArray
+
+const lastIndexArray =(array1, num1) => {
+return array1.lastIndexOf(num1)
+}
+
```
Comment (chrisyhlee) [inline code-challenges.js:105]: Great function! Indentation and spacing...

File: code-challenges.js [hunk 5]
```diff
@@ -9,61 +9,139 @@
...
 
 // Pseudo code:
+// input: array1
+// output: return largest to smallest numbers from array.
+// (possible solution is to create a function to use .sort((a, b) => a - b) method on array1 and use .reverse method to change the value form smallest to largest to largest to smallest then return the value)
+// function:sortArrayNumber
+
+const sortArrayNumber = (array1) => {
+return array1.sort((a, b) => a - b).reverse()
+}
```
Comment (chrisyhlee) [inline code-challenges.js:131]: How could you remove the need for `.reverse()` ?

**Review by chrisyhlee — APPROVED** [review]
Great work for assessment 1! This shows Advanced Beginner level! I left some comments for you to review, please look over them and reach out if you have any questions. I would focus on making sure your syntax is good to look at (indentation, space), and try to elaborate a little more in your pseudo code: explain some more about "how" the code is working!

File: code-challenges.js (hunk 5 above)
Comment (lanyuke) [reply code-challenges.js:131]: I think maybe  do this instead? :
const sortArrayNumber = (array1) => {
return array1.sort((a, b) => b - a)
}

I wasn't too sure how does the equation work....

**Pull request merged**